src/hooks/auth0-provider.js

```diff
@@ -89,7 +89,16 @@ const Auth0Provider = ({domain, clientId, children}) => {
   const clearSession = useCallback(
     async (...options) => {
       try {
-        await client.webAuth.clearSession(...options);
+        const {federated} = options.length ? options[0] : {};
+        const {credentialsManagerOnly} = options.length > 1 ? options[1] : {};
+        if (credentialsManagerOnly && federated) {
```

Comment:
@Widcket we don't expose the inbuilt CM instance because if we do that then we cannot handle the state management. So we abstract it through our Hooks API to keep the state in sync.

Reply:
> if we do that then we cannot handle the state management.

The developer in this scenario would not call `clearSession(`) but the CM's `clearCredentials()` instead, so it would not get out of sync/messed up.

Reply:
@Widcket in this library we do not integrate the CM for state management. Instead we have a higher level abstraction (Hooks) that does the state management. So if we clear the credentials directly using CM, the Hooks layer wouldn't know about it and will not end up updating it. That is why we need to do it at the Hooks layer to be in sync

Reply:
Let's discuss this in our sync tomorrow.

Reply:
Made the requested changes +1